### PR TITLE
Fix Sanity Test for Core Banned Functions

### LINKED ISSUE #15381: Stop Using Banned Core Functions
Guess what goes stale when you change all file extensions from .c to .cc?? [A python sanity linter that depends on the .c suffix](https://github.com/grpc/grpc/blob/master/tools/run_tests/sanity/core_banned_functions.py#L54)!!

```
LAPTOP ~/Desktop/grpc [master] (0) $ python tools/run_tests/sanity/core_banned_functions.py
Illegal use of "grpc_slice_unref(" in src/core/ext/filters/http/client_authority_filter.cc
Illegal use of "grpc_slice_ref(" in src/core/ext/filters/http/client_authority_filter.cc
Illegal use of "grpc_slice_unref(" in src/core/ext/transport/cronet/transport/cronet_transport.cc
Illegal use of "gpr_getenv_silent(" in src/core/lib/gpr/env_windows.cc
Illegal use of "gpr_getenv_silent(" in src/core/lib/gpr/log.cc
Illegal use of "gpr_getenv_silent(" in src/core/lib/gpr/env_linux.cc
Illegal use of "gpr_getenv_silent(" in src/core/lib/gpr/env_posix.cc
Illegal use of "grpc_slice_buffer_destroy(" in src/core/lib/transport/byte_stream.cc
Illegal use of "grpc_slice_unref(" in src/core/lib/security/security_connector/alts_security_connector.cc
Illegal use of "grpc_slice_buffer_destroy(" in src/core/lib/slice/slice_buffer.cc
Illegal use of "grpc_slice_buffer_reset_and_unref(" in src/core/lib/slice/slice_buffer.cc
Illegal use of "grpc_slice_unref(" in src/core/lib/slice/slice.cc
Illegal use of "grpc_slice_malloc(" in src/core/lib/slice/slice.cc
Illegal use of "grpc_slice_ref(" in src/core/lib/slice/slice.cc
Illegal use of "grpc_resource_quota_unref(" in src/core/lib/iomgr/resource_quota.cc
Illegal use of "grpc_resource_quota_ref(" in src/core/lib/iomgr/resource_quota.cc
Illegal use of "grpc_slice_unref(" in src/core/lib/iomgr/tcp_client_posix.cc
Illegal use of "grpc_error_create(" in src/core/lib/iomgr/error.cc
Illegal use of "grpc_error_unref(" in src/core/lib/iomgr/error.cc
Illegal use of "grpc_log_if_error(" in src/core/lib/iomgr/error.cc
Illegal use of "grpc_os_error(" in src/core/lib/iomgr/error.cc
Illegal use of "grpc_error_ref(" in src/core/lib/iomgr/error.cc
Illegal use of "grpc_wsa_error(" in src/core/lib/iomgr/error.cc
Illegal use of "grpc_slice_unref(" in src/core/tsi/ssl/session_cache/ssl_session_openssl.cc
Illegal use of "grpc_slice_malloc(" in src/core/tsi/ssl/session_cache/ssl_session_openssl.cc
Illegal use of "grpc_slice_unref(" in src/core/tsi/ssl/session_cache/ssl_session_cache.cc
Illegal use of "grpc_slice_unref(" in src/core/tsi/alts/zero_copy_frame_protector/alts_grpc_privacy_integrity_record_protocol.cc
Illegal use of "grpc_slice_unref(" in src/core/tsi/alts/handshaker/alts_tsi_event.cc
Illegal use of "grpc_slice_unref(" in src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc
Illegal use of "grpc_slice_ref(" in src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc
Illegal use of "grpc_slice_malloc(" in src/core/tsi/alts/handshaker/transport_security_common_api.cc
Illegal use of "grpc_slice_malloc(" in src/core/tsi/alts/handshaker/alts_handshaker_service_api.cc
Illegal use of "grpc_slice_unref(" in src/core/tsi/alts/handshaker/alts_tsi_utils.cc
Illegal use of "grpc_slice_unref(" in src/core/tsi/alts/handshaker/alts_handshaker_client.cc
Illegal use of "grpc_slice_unref(" in src/core/tsi/alts/handshaker/alts_handshaker_service_api_util.cc
Illegal use of "grpc_slice_malloc(" in src/core/tsi/alts/handshaker/alts_handshaker_service_api_util.cc
Traceback (most recent call last):
  File "tools/run_tests/sanity/core_banned_functions.py", line 63, in <module>
    assert errors == 0
AssertionError
```

### PR DESCRIPTION
**Pull request by ncteisen**
Fixes #15381

### CODE REVIEW

File: tools/run_tests/sanity/core_banned_functions.py [hunk 1]
```diff
@@ -61,3 +65,4 @@
                 errors += 1
 
 assert errors == 0
+assert num_files > 300  # we definitely have more than 300 files
```
Comment (yashykt) [inline tools/run_tests/sanity/core_banned_functions.py]: we probably do but why do we need to check that? :)

**Comment by ncteisen** [comment]
It's for when something else unexpected changes in the future and this
script stops checking any files. Just a hacky precaution

**Comment by yashykt** [comment]
Ah cool!

File: tools/run_tests/sanity/core_banned_functions.py (hunk 1 above)
Comment (yashykt) [reply tools/run_tests/sanity/core_banned_functions.py]: but maybe just add a comment for that so that no one removes it in the future

Comment (ncteisen) [reply tools/run_tests/sanity/core_banned_functions.py]: Good call, done

File: tools/run_tests/sanity/core_banned_functions.py [hunk 2]
```diff
@@ -65,4 +65,8 @@
                 errors += 1
 
 assert errors == 0
+# this check comes about from this issue:
+# https://github.com/grpc/grpc/issues/15381
+# basically, a change rendered this script useless and we did not realize.
```
Comment (yashykt) [inline tools/run_tests/sanity/core_banned_functions.py]: /s/basically/Basically

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++

 -+-+-+-+-+-+-+ MIXED  +-+-+-+-+-+-+-
  -0.0%     -64 [None]     +96  +0.0%

  -0.0%     -64 TOTAL      +96  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: tools/run_tests/sanity/core_banned_functions.py (hunk 2 above)
Comment (ncteisen) [reply tools/run_tests/sanity/core_banned_functions.py]: Done

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++

 -+-+-+-+-+-+-+ MIXED  +-+-+-+-+-+-+-
  -0.0%     -32 [None]     +96  +0.0%

  -0.0%     -32 TOTAL      +96  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by ncteisen** [comment]
Windows failure #15352
Bazel: infra flakes
Multi lang: #15405

**Pull request merged**